### DISCUSSION TITLE
Using Xbox buttons

### DISCUSSION THREAD

**Discussion by Synthix1** [pybricks/support#1573] [Q&A]
Hello pybricks!
First of all I need to give huge thumbs up for pybricks creators, I love it!

Could someone please help me with this, I belive simple, problem?
This code worked untill last update (I use beta)
Buttons A, B, LB and RB just does not work and causing a crash..

```
from pybricks.iodevices import XboxController
from pybricks.parameters import Direction, Port
from pybricks.pupdevices import Motor
from pybricks.pupdevices import DCMotor

# Set up all devices.
left = Motor(Port.B, Direction.COUNTERCLOCKWISE)
right = Motor(Port.C, Direction.COUNTERCLOCKWISE)
arm = Motor(Port.A)
hand = DCMotor(Port.D)
controller = XboxController()


# The main program starts here.
while True:


    # Steer with the left joystick.
    lefthorizontal, leftvertical = controller.joystick_left()
    righthorizontal, rightvertical = controller.joystick_right()
    lefttrigger, righttrigger = controller.triggers()
    pressed = controller.buttons.pressed()
    
    left.dc((0.6*leftvertical)+righttrigger-lefttrigger )
    right.dc((0.6*rightvertical)+righttrigger-lefttrigger)

    if Button.A in pressed:
        arm.dc(100)
    elif Button.B in pressed:
        arm.dc(-100)
    else:
        arm.dc(0)

    if Button.LB in pressed:
        hand.dc(100)
    elif Button.RB in pressed:
        hand.dc(-100)
    else:
        hand.dc(0)
```

**Comment by laurensvalk** [comment]
> Buttons A, B, LB and RB just does not work..

Use `Button.A` instead of "A", and so on. This was changed in an early beta version. 

> First of all I need to give huge thumbs up for pybricks creators, I love it!

Thank you! Much appreciated. :smile:

**Reply by Synthix1** [reply]
Thank you for the answer. 
I've already tried it moment ago (and also edited the code above)
Still no luck. Don´t I have some syntax problem?

**Reply by laurensvalk** [reply]
Yes, but it's related. The `Button` is missing from the `import`. Try:

```python
from pybricks.parameters import Direction, Port, Button
```

**Reply by Synthix1** [reply]
It works. Thank you very much!

**Reply by laurensvalk** [reply]
But what did you build? :open_mouth:

**Reply by Synthix1** [reply]
Highly improvised video of the creation here:
https://youtu.be/xGGmgfF0F0k

No animals were harmed..

Tank-like driven vehicle with robust caster wheel, arm for up and down and hand for grabbing.. 
It is intended to collect kid's toys from the floor and putting them to a toy box. 
I'm building sort of a conveyor/lift to reach higher ground, so I'm looking forward for the HUB to HUB communication. 
I will share the result ;-)

**Reply by laurensvalk** [reply]
Well done! Looks great :)
